.github/workflows/test_dockerfiles.yaml

```diff
@@ -0,0 +1,96 @@
+name: Dockerfile tests
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - '**/Dockerfile'
+  schedule:
+    - cron: '23 2 * * 0'
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+jobs:
+  define_changed_dockerfile_matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.changed-dockerfiles.outputs.all_changed_files }}
+    steps:
+      - uses: actions/checkout@v4
+      -
+        name: Get Changed Dockerfiles
+        id: changed-dockerfiles
+        uses: tj-actions/changed-files@v44
+        with:
+          matrix: true
+          files: '**/Dockerfile'
+          files_ignore: '**/GlimpseImputationPipeline/glimpse_docker/Dockerfile'
+      - name: List dockerfiles
+        run: echo '${{ steps.changed-dockerfiles.outputs.all_changed_files}}'
+  
+  define_all_dockerfile_matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.all-dockerfiles.outputs.files}}
+    steps:
+      - uses: actions/checkout@v4
+      -
+        name: Get all Dockerfiles
+        id: all-dockerfiles
+        run: |
+          {
+            echo 'files<<EOF'
+            find . -name "Dockerfile" ! -path "*/GlimpseImputationPipeline/glimpse_docker/*" | jq -R . | jq -s .
+            echo EOF
+          } >> "$GITHUB_OUTPUT"
+      - name: List dockerfiles
+        run: echo '${{ steps.all-dockerfiles.outputs.files}}'
+
+  test_builds:
+    runs-on: ubuntu-latest
+    needs: [define_changed_dockerfile_matrix, define_all_dockerfile_matrix]
+    strategy:
+      matrix:
+        dockerfile: ${{ github.ref != 'refs/heads/main' && fromJSON(needs.define_changed_dockerfile_matrix.outputs.matrix) || fromJSON(needs.define_all_dockerfile_matrix.outputs.matrix) }}
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v4
+      -
+        name: Get context
+        id: get-context
+        run: |
+            echo "context=$(dirname ${{ matrix.dockerfile }})" >> $GITHUB_OUTPUT
+      -
+        name: Build
+        uses: docker/build-push-action@v6
+        with:
+          context: ${{ steps.get-context.outputs.context }}
+          file: ${{ matrix.dockerfile }}
+
+  test_glimpse_docker:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      -
+        name: Check if Dockerfile has changed
+        id: check-if-changed
+        if: ${{ github.ref != 'refs/heads/main' }}
+        uses: tj-actions/changed-files@v44
+        with:
+          files: '**/GlimpseImputationPipeline/glimpse_docker/Dockerfile'
+      -
+        name: Build main GLIMPSE Docker
```

Comment:
What's the purpose of testing the build of the docker image? Presumably to test any changes made in the pipeline, one would have to build the docker, upload it, and run the WDL already, right?

Reply:
I'm not sure I follow?  All of these docker tests are just to make sure they build, not to test whether the build work correctly in the pipelines.  The one difference here is that we need to build the glimpse tool docker on it's own and then build the glimpse pipeline docker from that.  I believe the reason for doing that was basically to not have to duplicate the glimpse tool docker in our pipeline docker, but still be able to add things to it.

Reply:
I agree testing that an updated docker actually does what you want would be even better, but I think that would require automatic build-push (easy) and associated wdl update (more complex but intriguing).  For the moment I'm just thinking of this as both a check to catch obvious mistakes, and a check against external changes breaking our Dockerfiles.